### PR TITLE
document object_ha (donkeys)

### PR DESCRIPTION
**Pull request by MegaIDK**
i didnt mess with the `limbIndex`es cus `z_en_horse.c` is used by other stuff like epona apparently
also it brought commits from the other branch as well?? wat (edit: fixed)

### CODE REVIEW

**Review by jpburnett — COMMENTED** [review]
Any particular reason for calling them donkeys instead of horses?

**Comment by tom-overton** [comment]
> Any particular reason for calling them donkeys instead of horses?

Cremia's animal is almost certainly intended to be a donkey or mule:
differences
image

However, this same image can be used to argue the Gorman Bros.'s animals are horses, and in fact, they are referred to as such in-game too. The Bomber's Notebook calls them:
> "Gorman Brothers
Horse trainers. Suspicious..."

And Gorman (the middle child in town) says:
> "Anyway, I'm just the despised
second son! I can't handle horses like my older
brother! I can't take care of horses like
my younger brother!"

So it looks like this object has one donkey and one horse? That's pretty annoying, though.

**Review by jpburnett — APPROVED** [review]
That'll do donkey, that'll do

File: assets/xml/objects/object_ha.xml [hunk 1]
```diff
@@ -1,135 +1,162 @@
...
+        <DList Name="gDonkeyCremiaRightFrontForearm" Offset="0x116A8" />
+        <DList Name="gDonkeyCremiaRightFrontLegRoot" Offset="0x11888" />
+        <DList Name="gDonkeyCremiaHead" Offset="0x11948" />
+        <DList Name="gDonkeyCremiaUpperNeck" Offset="0x11E08" />
+        <DList Name="gDonkeyCremiaLowerNeck" Offset="0x11F90" />
+        <DList Name="gDonkeyCremiaLeftFrontFoot" Offset="0x122C0" />
+        <DList Name="gDonkeyCremiaLeftFrontCannon" Offset="0x12460" />
+        <DList Name="gDonkeyCremiaLeftFrontForearm" Offset="0x125E8" />
+        <DList Name="gDonkeyCremiaLeftFrontLegRoot" Offset="0x127D0" />
+        <DList Name="gDonkeyCremiaTorso" Offset="0x12890" />
```
Comment (AngheloAlf) [inline assets/xml/objects/object_ha.xml:120]: This displaylists are missing the `DL` suffix

**Review by AngheloAlf — COMMENTED** [review]
After the donkey discussion I think it would be better to rename the gorman one to "horse"

File: assets/xml/objects/object_ha.xml [hunk 2]
```diff
@@ -1,135 +1,162 @@
...
-        <Limb Name="object_ha_Standardlimb_015020" Type="Standard" EnumName="OBJECT_HA_2_LIMB_13" Offset="0x15020" />
-        <Limb Name="object_ha_Standardlimb_01502C" Type="Standard" EnumName="OBJECT_HA_2_LIMB_14" Offset="0x1502C" />
-        <Limb Name="object_ha_Standardlimb_015038" Type="Standard" EnumName="OBJECT_HA_2_LIMB_15" Offset="0x15038" />
-        <Limb Name="object_ha_Standardlimb_015044" Type="Standard" EnumName="OBJECT_HA_2_LIMB_16" Offset="0x15044" />
-        <Limb Name="object_ha_Standardlimb_015050" Type="Standard" EnumName="OBJECT_HA_2_LIMB_17" Offset="0x15050" />
-        <Limb Name="object_ha_Standardlimb_01505C" Type="Standard" EnumName="OBJECT_HA_2_LIMB_18" Offset="0x1505C" />
-        <Limb Name="object_ha_Standardlimb_015068" Type="Standard" EnumName="OBJECT_HA_2_LIMB_19" Offset="0x15068" />
-        <Skeleton Name="object_ha_Skel_0150D8" Type="Flex" LimbType="Standard" LimbNone="OBJECT_HA_2_LIMB_NONE" LimbMax="OBJECT_HA_2_LIMB_MAX" EnumName="ObjectHa2Limb" Offset="0x150D8" />
+
+        <Animation Name="gDonkeyUnusedAnim" Offset="0xC4" /> <!-- Original name is "ha_default" -->
```
[the remark this answers is not in the document]
Comment (MegaIDK) [reply assets/xml/objects/object_ha.xml:5]: iirc its just like 2 frames of the model standing still its not even animated or anything

**Comment by MegaIDK** [comment]
just `gHorse` or `gHorseBandit`?

**Comment by tom-overton** [comment]
I would go with `gHorseBandit` to distinguish it from any other horses, personally.

File: assets/xml/objects/object_ha.xml (hunk 2 above)
[the remark this answers is not in the document]
Comment (tom-overton) [reply assets/xml/objects/object_ha.xml]: For humanoid characters, animations called "default" or "kihon" (Japanese for "basics; fundamentals; basis; foundation​") are the equivalent of a T-pose. They aren't always *doing* a T-pose (I think Odolwa's arms are at his sides, for example), but it's clear the these animations are effectively the "base" pose of the model. So this `ha_default` animation is something similar, but I have no idea what to name it, especially for horses/donkeys.

**Review by hensldm — APPROVED** [review]

**Review by AngheloAlf — APPROVED** [review]

**Review by engineer124 — APPROVED** [review]

**Pull request merged**